contracts/distribution/PushDistributor.sol

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.8.20;
+
+import { Party } from "./../party/Party.sol";
+import { PartyGovernance } from "./../party/PartyGovernance.sol";
+import { ProposalExecutionEngine } from "./../proposals/ProposalExecutionEngine.sol";
+import { IERC20 } from "../tokens/IERC20.sol";
+import { LibSafeCast } from "../utils/LibSafeCast.sol";
+import { LibERC20Compat } from "./../utils/LibERC20Compat.sol";
+import { LibAddress } from "./../utils/LibAddress.sol";
+
+contract PushDistributor {
+    event Distributed(Party party, IERC20 token, address[] members, uint256 amount);
+
+    error NotEnoughETH(uint256 expectedAmount, uint256 receivedAmount);
+    error UnexpectedETH(uint256 amount);
+    error WrongProposalId(uint256 proposalId);
+    error WrongMembers();
+    error MembersNotSorted();
+
+    using LibSafeCast for uint256;
+    using LibERC20Compat for IERC20;
+    using LibAddress for address payable;
+
+    // Token address used to indicate ETH.
+    IERC20 private constant ETH_ADDRESS = IERC20(0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE);
+
+    function distribute(
+        IERC20 token,
+        address[] memory members,
+        uint256 amount,
+        uint256 proposalId
+    ) external payable {
+        Party party = Party(payable(msg.sender));
+        if (token == ETH_ADDRESS && msg.value < amount) revert NotEnoughETH(amount, msg.value);
+
```

Comment:
`UnexpectedETH` no longer emitted if its an ERC20 distribution

Reply:
I'm aware, it's not necessary anymore since we return any remaning ETH at the end.